src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/ConfigField.java

```diff
@@ -34,27 +41,66 @@
     private boolean sensitive;
     private FieldGroup group;
     private String subGroup;
+    private transient BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction;
 
-    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group, final String subGroup) {
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group, final String subGroup,
+        final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
         this.key = key;
         this.label = label;
         this.type = type;
         this.required = required;
         this.sensitive = sensitive;
         this.group = group;
         this.subGroup = subGroup;
+        this.validationFunction = validationFunction;
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group) {
-        this(key, label, type, required, sensitive, group, "");
+        this(key, label, type, required, sensitive, group, "", null);
+    }
+
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group, final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
+        this(key, label, type, required, sensitive, group, "", validationFunction);
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final String subGroup) {
-        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup);
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup, null);
```

Comment:
I would feel better about a storing a private constant with a lambda to pass here instead of null. If we are already going to be applying validation to fields, we could do something like this: 
`private static final BiFunction<FieldValueModel, FieldModel, Collection<String>> NO_VALIDATION = (x, y) -> List.of();`
I'm not 100% clear on how we want to do validation with default options, but something along these lines seems safer, though it will add more operations.